src/main/java/com/diyasylum/formfiller/models/I589Field.java

```diff
@@ -0,0 +1,94 @@
+package com.diyasylum.formfiller.models;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.Objects;
+import org.apache.pdfbox.pdmodel.interactive.form.PDField;
+
+public class I589Field {
+  private final String description;
+  private final String absolutePath;
+  private final String relativePath;
+  private final String value;
+  private final FieldType fieldType;
+
+  public static I589Field fromPdField(PDField field) {
+    return new I589Field(
+        field.getAlternateFieldName(),
+        field.getFullyQualifiedName(),
+        field.getPartialName(),
+        FieldType.fromString(field.getFieldType()),
+        field.getValueAsString());
+  }
+
+  @JsonCreator
+  public I589Field(
```

Comment:
Since these properties are specific to the internal mumbo jumbo of the PDF, I assume part of your next task is to make a more intuitive interface that allows to pass variables like "applicant first name."

Reply:
Its gonna take a few iterations. But over time im expecting this to get more intuitive. With this many fields its gonna be a bit...

Reply:
but yeah, my plan is to do structure from the top down... working my way down to the individual fields.